packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -535,6 +569,8 @@ export class ApolloServerBase {
   protected async graphQLServerOptions(
     integrationContextArgument?: Record<string, any>,
   ) {
+    await this.createSchemaDerivedDataPromise;
```

Comment:
Is this particular `await` necessary because of the places where `runHttpQuery` is called directly?

Reply:
I put the await there because the method returns a thing that depends on `this.schema`, and we don't know if `this.schema` will be defined until that promise resolves. Said elsewhere, but worth looking into how we can make such use-before-definition errors impossible to represent.